nautobot/ipam/api/serializers.py

```diff
@@ -321,27 +321,30 @@ class Meta:
         fields = "__all__"
 
 
-class PrefixLengthSerializer(serializers.Serializer):
-    prefix_length = serializers.IntegerField()
-
-    def to_internal_value(self, data):
-        requested_prefix = data.get("prefix_length")
-        if requested_prefix is None:
-            raise serializers.ValidationError({"prefix_length": "this field can not be missing"})
-        if not isinstance(requested_prefix, int):
-            raise serializers.ValidationError({"prefix_length": "this field must be int type"})
-
-        prefix = self.context.get("prefix")
-        if prefix.ip_version == 4 and requested_prefix > 32:
-            raise serializers.ValidationError({"prefix_length": f"Invalid prefix length ({requested_prefix}) for IPv4"})
-        elif prefix.ip_version == 6 and requested_prefix > 128:
-            raise serializers.ValidationError({"prefix_length": f"Invalid prefix length ({requested_prefix}) for IPv6"})
-        return data
+class PrefixLengthSerializer(PrefixLegacySerializer):
+    """
+    Input serializer for POST to /api/ipam/prefixes/<id>/available-prefixes/, i.e. allocating one or more sub-prefixes.
+
+    Since setting of multiple locations on create is not supported, this uses the legacy single-location option.
+    """
+
+    prefix_length = serializers.IntegerField(required=True)
+
+    class Meta(PrefixLegacySerializer.Meta):
+        fields = PrefixLegacySerializer.Meta.fields.copy()
```

Comment:
I would guess not, but just double checking deep copy would not be more appropriate.

Reply:
It's just an array of strings so I'm pretty sure copy is OK. Thanks for checking!